python/nx-cugraph/_nx_cugraph/__init__.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tell NetworkX about the cugraph backend. This file can update itself:
+
+$ make plugin-info  # Recommended method for development
+
+or
+
+$ python _nx_cugraph/__init__.py
+"""
+
+# Entries between BEGIN and END are automatically generated
+_info = {
+    "backend_name": "cugraph",
+    "project": "nx-cugraph",
+    "package": "nx_cugraph",
+    "url": "https://github.com/rapidsai/cugraph/tree/branch-23.10/python/nx-cugraph",
+    "short_summary": "GPU-accelerated backend.",
+    # "description": "TODO",
+    "functions": {
+        # BEGIN: functions
+        "betweenness_centrality",
+        "edge_betweenness_centrality",
+        "louvain_communities",
+        # END: functions
+    },
+    "extra_docstrings": {
+        # BEGIN: extra_docstrings
+        "betweenness_centrality": "`weight` parameter is not yet supported.",
+        "edge_betweenness_centrality": "`weight` parameter is not yet supported.",
+        "louvain_communities": "`threshold` and `seed` parameters are currently ignored.",
+        # END: extra_docstrings
+    },
+    "extra_parameters": {
+        # BEGIN: extra_parameters
+        "louvain_communities": {
+            "max_level : int, optional": "Upper limit of the number of macro-iterations.",
+        },
+        # END: extra_parameters
+    },
+}
+
+
+def get_info():
+    """Target of ``networkx.plugin_info`` entry point.
+
+    This tells NetworkX about the cugraph backend without importing nx_cugraph.
+    """
+    # Convert to e.g. `{"functions" {"myfunc": {"extra_docstrings": ...}}}`
+    d = _info.copy()
+    func_keys = ("extra_docstrings", "extra_parameters")
+    d["functions"] = {
+        func: {key: vals[func] for key in func_keys if func in (vals := d[key])}
+        for func in d["functions"]
+    }
+    for key in func_keys:
+        del d[key]
+    return d
+
+
+__version__ = "23.10.00"
```

Comment:
Should [update-version.sh](https://github.com/rapidsai/cugraph/blob/branch-23.10/ci/release/update-version.sh) be updated for this?

Reply:
Good catch! Done.